### PR TITLE
feat: proxy events

### LINKED ISSUE #183: Python proxy events
Let's implement proxy upload events and job to check the packages and add them to db.

### PR DESCRIPTION
**Pull request by olenagerasimova**
closes #183
Implemented proxy events queue and job to check packages and them to packages queue. The job itself will be started from artipie/artipie module if packages db is required and configured in repository settings.

### CODE REVIEW

File: src/main/java/com/artipie/pypi/PyProxyPackageProcessor.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                if (event != null) {
+                    final Key key = event.artifactKey();
+                    final String filename = new KeyLastPart(key).get();
+                    this.asto.value(key).thenCompose(
+                        val -> new ContentAsStream<PackageInfo>(val).process(
+                            input -> new Metadata.FromArchive(input, filename).read()
+                        )
+                    ).thenCompose(
+                        info -> {
+                            final CompletionStage<Void> res;
```
Comment (dgarus) [inline src/main/java/com/artipie/pypi/PyProxyPackageProcessor.java:68]: Do we really need to create a future here?
We are already in the separate thread, and we want to finish the job here, that is why we call `join()`. So, I think we could just do all work in sync manner.

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks!
Please read my comment

**Comment by olenagerasimova** [comment]
@dgarus thanks, corrected

File: src/main/java/com/artipie/pypi/http/PySlice.java [hunk 2]
```diff
@@ -43,7 +45,7 @@ public final class PySlice extends Slice.Wrap {
      * @param storage The storage and default parameters for free access.
      */
     public PySlice(final Storage storage) {
-        this(storage, Policy.FREE, Authentication.ANONYMOUS, "*", new EventQueue<>());
+        this(storage, Policy.FREE, Authentication.ANONYMOUS, "*", Optional.of(new LinkedList<>()));
```
Comment (dgarus) [inline src/main/java/com/artipie/pypi/http/PySlice.java:48]: `LinkedList` is not thread safe collection, but it is changed inside futures. It doesn't look correct.
Do we really need to create a queue here? Can we pass `Optional.empty()`?

**Review by dgarus — APPROVED** [review]
Thanks! lgtm

**Pull request merged**